src/target/riscv/riscv.c

```diff
@@ -1914,13 +1918,14 @@ static int riscv_resume(
 static int riscv_target_resume(struct target *target, int current,
 		target_addr_t address, int handle_breakpoints, int debug_execution)
 {
-	LOG_TARGET_DEBUG(target, "target->state=%d, current=%s, address=0x%"
-		TARGET_PRIxADDR ", handle_breakpoints=%s, debug_exec=%s",
-		target->state,
-		current ? "true" : "false",
-		address,
-		handle_breakpoints ? "true" : "false",
-		debug_execution ? "true" : "false");
+	/* FIXME: It would be better to replace this `if` with an `assert`
+	 * by moving it to the target-independent part of resume command handler.
+	 * Please, see https://review.openocd.org/c/openocd/+/7750
+	 */
+	if (target->state != TARGET_HALTED) {
```

Comment:
I'd recommend to leave here the "if" (instead of an assert) and drop the comment.
I prefer the "if" for the sake of robustness - that's because we have less control over the common, target independent code, and it can change over time.

```suggestion
	if (target->state != TARGET_HALTED) {
```

As for https://review.openocd.org/c/openocd/+/7750 - overall it looks good but as mentioned already in the upstream review, I don't know enough about other targets to be able to confirm that change.

Reply:
Addressed.